### ISSUE TITLE
Question about the model

### ISSUE THREAD

**Issue by rodyt** [yukihiko/ThreeDPoseUnitySample#10]
Hello,

Great work on 3D pose estimation. It's truly amazing what ML can do these days! I was wondering about a couple of things about your model:

1. What model architecture is used for the ML model? (hourglass, cpm, etc...)
2. How was the model trained? (heatmap regression; if so, how?; loss function, optimizers, lr?)
3. What dataset(s) was this model trained on?
4. For the iOS model, what set of mobile optimizations were performed on it to get it to run on a mobile device (quantization?) Does your phone heat up when you run it?

It also appears that this model is an older version. On your twitter account, it seems that that model has higher accuracy and is newer. What modifications did you do to make it perform better?

Thank you

**Comment by yukihiko** [comment]
Hello

1. The model architecture used in this project is based on MobileNet. I have modified it a little, but I don't think it was effective.
2. The output is a 2D and 3D heat map and offset from it, and the loss is the mean square. I am referring to PoseNet.
3.The data set for 2D pose estimation was LSP, MPII, and COCO. Ultimately, COCO is the main.
I created the 3D dataset myself. I use Unity 3D characters.
4. When moving with swift, CoreML. ONNX when using Unity. These models have only been converted. I haven't used quantization yet.

In the new model, there is a change in the calculation method of the heat map, a change to ResNet. The change to ResNet was effective, but it became larger and slower. In ResNet50, real-time execution has become stricter.
Now I'm trying a model with a larger input image size.

**Comment by rodyt** [comment]
Thank you for replying to my question.

I'm still a little curious about a few things:

1. How does your model work? My understanding is it first regresses 2D pose and then lifts it up to 3D (i.e. First predicts 2D heatmaps (Batch Size, Height, Width, Joints) which are fed into more convolution layers that output a 5D tensor of (Batch Size, Height, Width, Depth, Joints), which takes a loss across all 3 axes using MSE). Is this correct? 

Also, are the annotations for the 3D pose relative to the hip (i.e. hip is (0, 0, 0)) or are they absolute positions within the 3D heatmap coordinate space? How long did it take for you to create the dataset (it seems you must have worked hard and dedicated a lot of time to it)?

2. How did you train the model? Did you train the 2D part independently & 3D part independently?

3. In your twitter posts, you mention how accuracy suffers if the person is not fully visible in the frame. How do you go about solving this constraint?

I wonder how Apple manages to perform pose estimation in their ARKit3...

What you're doing is really cool, so keep up with the great work. Is there anybody else you know who's doing similar things like this?

Thank you for being so active and helpful!

**Comment by yukihiko** [comment]
1. No. 2D dataset estimates a 2D heat map, and a 3D dataset directly estimates a 3D heat map. The output of 3D pose does not use 2D information. Since the image domain is biased only in the 3D data set created by Unity, general feature points can be learned from the 2D data set of COCO. (I'm sorry, I can't express well in English)
Relative coordinates from the abdomen. The main coding is a few days. After a few days, I increased the number of characters little by little. I am still modifying it to increase the number of poses. It takes about an hour to create 10,000 data automatically. Now I use about 100,000 data.
2. Simultaneous.
3. When the joint is not visible in the image, the score of the heat map is low, so in that case, I think that it will be a little better if the motion is fixed with that vector. But I want you to move when the joint is hidden in body. ArKit3 looks similar.
Some people use open pose.

**Comment by littleqiang-go** [comment]
OpenCVException: Failed to parse onnx model
OpenCvSharp.NativeMethods+<>c.<.cctor>b__1805_0 (OpenCvSharp.ErrorCode status, System.String funcName, System.String errMsg, System.String fileName, System.Int32 line, System.IntPtr userdata) (at <1c4180998434408a87e8d3a3d5f215e6>:0)
OpenCvSharp.Dnn.Net.ReadNetFromONNX (System.String onnxFile) (at <1c4180998434408a87e8d3a3d5f215e6>:0)
ThreeDPoseScript.InitONNX () (at Assets/Scenes/ThreeDPoseScript.cs:379)
ThreeDPoseScript.Start () (at Assets/Scenes/ThreeDPoseScript.cs:158)

I run the project in windows, but it show a error, can you help me

**Comment by yukihiko** [comment]
I wonder why? I think it works on Windows. x64, right?

Is this thread okay??

**Comment by xinghuokang** [comment]
Hi, Amazing demo you provide, Can you provide detailed engineering procedures, complete test process?

**Comment by DukeZacharia** [comment]
Please I have the same problem but I fixed it. When I still try to run it, it crashes without providing an error please what may be the main problem?
I am running it on windows.
